src/lib/structures/Argument.ts

```diff
@@ -36,72 +35,107 @@ export interface ArgumentChoice {
 export interface ArgumentOptions {
 	name: string;
 	description: string;
-	type: ArgumentType | keyof typeof ArgumentType | ApplicationCommandOptionType | keyof typeof ApplicationCommandOptionType;
+	type:
+		| ArgumentType
+		| keyof typeof ArgumentType
+		| ApplicationCommandOptionType
+		| keyof typeof ApplicationCommandOptionType;
 	required?: boolean;
 	choices?: Array<ArgumentChoice>;
-	options?: Array<Argument>;
+	arguments?: Array<Argument | ArgumentOptions>;
+	/**
+	 * @deprecated Please use ArgumentOptions#arguments instead
+	 * @link https://garlic-team.js.org/docs/#/docs/gcommands/next/typedef/ArgumentOptions
+	 */
+	options?: Array<Argument | ArgumentOptions>;
 	channelTypes?: Array<ChannelType | keyof typeof ChannelType>;
 	run?: (ctx: AutocompleteContext) => any;
 }
 
+const validationSchema = z
+	.object({
+		name: z
+			.string()
+			.max(32)
+			.regex(/^[a-zA-Z1-9]/),
+		description: z.string().max(100),
+		type: z
+			.union([z.string(), z.nativeEnum(ArgumentType)])
+			.transform((arg) =>
+				typeof arg === 'string' && Object.keys(ArgumentType).includes(arg) ? ArgumentType[arg] : arg,
+			),
+		required: z.boolean().optional(),
+		choices: z
+			.object({
+				name: z.string(),
+				value: z.string(),
+			})
+			.array()
+			.optional(),
+		arguments: z.any(),
```

Comment:
Why `.any()`? There can be `z.object({ name: z.string(), description: z.string() })` etc

Reply:
Recursion is an issue with zod (for subcommands), so to prevent a lot of duplicate code, it's validated at the creation of the Argument class. See Argument.ts (Argument instance is created if the argument is not one already)